source/material_model/rheology/compositional_viscosity_prefactors.cc

```diff
@@ -0,0 +1,142 @@
+/*
+  Copyright (C) 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/rheology/compositional_viscosity_prefactors.h>
+#include <aspect/utilities.h>
+#include <aspect/global.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+#include <aspect/simulator_signals.h>
+#include <aspect/parameters.h>
+
+
+// TODO: Add constant_viscosity_prefactors to this file at a point when ASPECT is ready to be broken
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      CompositionalViscosityPrefactors<dim>::CompositionalViscosityPrefactors ()
+        = default;
+
+      template <int dim>
+      double
+      CompositionalViscosityPrefactors<dim>::compute_viscosity (const MaterialModel::MaterialModelInputs<dim> &in,
+                                                                const double base_viscosity,
+                                                                const unsigned int composition_index,
+                                                                const unsigned int q) const
+      {
+        double prefactor;
+        switch (viscosity_prefactor_scheme)
+          {
+            case none:
+            {
+              prefactor = 1;
+              break;
+            }
+            case water_fugacity:
+            {
+              const unsigned int bound_fluid_idx = this->introspection().compositional_index_for_name("bound_fluid");
+              const double wt_fraction_h2o = in.composition[q][bound_fluid_idx];
+              const double wt_ol = 1 - wt_fraction_h2o; // We need to get wt_h2o from bound water
```

Comment:
Should this by `wt_fraction_ol`? I would also rename this to `wt_fraction_olivine`. For additional clarity, I may even rename `wt` to `weight`